nixos/doc/manual/development/option-types.section.md

```diff
@@ -202,6 +202,12 @@ merging is handled.
 
 :   A string. Multiple definitions are concatenated with a colon `":"`.
 
+`types.hexStr`
+
+:   A hexadecimal value represented as string. The type only allows lowercase
```

Comment:
```suggestion
:   A sequence of bytes represented as a string of hexadecimals. The type only allows lowercase
```

Reply:
>  Half bytes seem like a mistake we should catch.

Uh, I think we might be talking past each other. I this is a string of hex (as in characters representing hex numbers) and not a byte sequence of hex data. So you don't really have half bytes :eyes:

Reply:
Yes, I did make somewhat of a leap, and I'm sorry if my presentation was messy. I'm assuming half bytes are not ok, and I think the type should check that.

Do you have a use case where the string of hexadecimal digits does not represent bytes?

Reply:
I don't think it would be beneficial to talk about half-bytes here. Just saying hexadecimal sounds good

Reply:
I agree half bytes are unnecessarily awkward, and I suspect they should not be allowed. If we don't allow them, we can call this a sequence of bytes in hexadecimal; no silly language, and no silly values.
If we do have a use case for odd numbers of hex digits *with* leading zeroes, then something like "hexadecimal string" sounds good to me.

Reply:
Hmm, maybe I wasn't communicating well, because I was trying to comment about the implementation, but my suggestions are on lines about documentation.
I like to use file comments because they become threads, but it seems that I need to take into account better where I put those comments.

Reply:
I'm also only getting it now: You're hinting at wanting a check to prevent against an odd number of hex digits (which would represent half-bytes), which is most likely just a bug. I've personally never encountered any programs asking for an odd number of hex digits (because indeed, hex digits are usually used to represent bytes).

I think it's fair to restrict this type to an even number of hex digits to prevent such mistakes. We can still allow an odd number of digits later if this turns out that there are use cases for an odd number of digits. Because of that though I wouldn't want this to be embedded into the documentation. So I think something like this should be good:

```suggestion
:   A string of hexadecimal digits. The type only allows lowercase
```